openedx/core/djangoapps/user_authn/views/logout.py

```diff
@@ -153,8 +152,8 @@ def get_context_data(self, **kwargs):
             'target': target,
             'logout_uris': logout_uris,
             'enterprise_target': self._is_enterprise_target(target),
-            'saml_logout_url': KEY_CLOCK_LOGOUT_URL.format(
-                self.saml_auth_backend.get_idp(self.third_party_provider.slug.encode()).conf['entity_id'].encode(),
+            'saml_logout_url': '{}?/redirect_uri={}'.format(
+                json.loads(self.third_party_provider.other_settings)['logout_url'].encode(),
```

Comment:
I think there should a way to get config via python object instead of converting dict to json and then pulling it from json.

Reply:
In Python object `<SAMLProviderConfig: SAMLProviderConfig object>` other_settings key is type of string. that's why need to convert into JSON to get `logout_url`